timeseries/src/main/scala/com/criteo/cuttle/timeseries/TimeSeriesScheduler.scala

```diff
@@ -723,16 +749,37 @@ case class TimeSeriesScheduler(logger: Logger,
       case (_, _, effect) => effect.isCompleted
     }
 
-    val (stateSnapshot, completedBackfills, toRun) = atomic { implicit txn =>
+    val (stateSnapshot, completedBackfills, toRun, debounced, commitInstant) = atomic { implicit txn =>
+      val now = Instant.now
       val (stateSnapshot, newBackfills, completedBackfills) =
         collectCompletedJobs(_state(), _backfills(), completed)
 
-      val toRun = jobsToRun(workflow, stateSnapshot, Instant.now, executor.projectVersion)
+      val (toRun, debounced) = jobsToRun(workflow, stateSnapshot, now, executor.projectVersion)
 
       _state() = stateSnapshot
       _backfills() = newBackfills
 
-      (stateSnapshot, completedBackfills, toRun)
+      (stateSnapshot, completedBackfills, toRun, debounced, now)
+    }
+
+    toRun.groupBy(_._1).foreach {
+      case (job, _) =>
+        if (job.scheduling.batching.size > 1) {
+          logger.debug(s"scheduling batches for ${job.name}")
+          debouncedJobs -= job
+        }
+    }
+
+    debounced.groupBy(_._1).foreach {
+      case (job, _) =>
+        debouncedJobs.contains(job) match {
```

Comment:
OPT: debouncedJobs.get() match { Some(x) => ... } to avoid the debouncedJobs(job) afterwards

Reply:
done